File: android/BertSpeak/app/src/main/java/chuckcoughlin/bert/tab/BasicAssistantFragment.java
/**
 * Copyright 2019 Charles Coughlin. All rights reserved.
 *  (MIT License)
 */
package chuckcoughlin.bert.tab;


import android.os.Bundle;
import android.support.v4.app.Fragment;

/**
 * This is a base class for all the page fragments, except lists.
 */

public class BasicAssistantFragment extends Fragment implements AssistantFragment {

    /**
     * A no-arg constructor is required.
     */
    public BasicAssistantFragment() {
        Bundle bundle = new Bundle();
        setArguments(bundle);
    }


    public int getPageNumber() {
        return getArguments().getInt(PAGE_ARG);
    }
    public void setPageNumber(int page) {
        getArguments().putInt(PAGE_ARG, page);
    }
    public String getTitle()   {
        return getArguments().getString(TITLE_ARG);
    }
    public void setTitle(String title)   {
        getArguments().putString(TITLE_ARG, title);
    }

    /**
     * The saved state becomes the fragment's state.
     */
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    /**
     * Everything that matters is in the bundle.
     * @param stateToSave dictionary of values to persist
     */
    @Override
    public void onSaveInstanceState(Bundle stateToSave) {
        super.onSaveInstanceState(stateToSave);
    }
}

File: BertSpeak/app/src/main/kotlin/chuckcoughlin/bertspeak/tab/BasicAssistantFragment.kt
/**
 * Copyright 2022 Charles Coughlin. All rights reserved.
 * (MIT License)
 */
package chuckcoughlin.bertspeak.tab

import android.os.Bundle
import androidx.fragment.app.Fragment


/**
 * This is a base class for all the page fragments, except lists. "open" implies extendable.
 */
open class BasicAssistantFragment(page:Int): Fragment(), AssistantFragment {
    override val pageNumber = page
    override var title: String?
        get() {
            assert(arguments != null)
            return arguments?.getString(AssistantFragment.Companion.TITLE_ARG)
        }
        set(title) {
            assert(arguments != null)
            arguments?.putString(AssistantFragment.Companion.TITLE_ARG, title)
        }

    /**
     * Everything that matters is in the bundle.
     * @param stateToSave dictionary of values to persist
     */
    override fun onSaveInstanceState(stateToSave: Bundle) {
        super.onSaveInstanceState(stateToSave)
    }

    /**
     * A no-arg constructor is required.
     */
    init {
        val bundle = Bundle()
        arguments = bundle
    }
}
